pipeline.sh

```diff
@@ -9,8 +9,16 @@
 #SBATCH --export=NONE
 #SBATCH --get-user-env=L60
 
-SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
-source "${SCRIPT_DIR}"/utils/header.sh
+# Retrieve original directory of submitted script.
+if [ -n "$SLURM_JOB_ID" ] ; then # If Slurm job.
+  SCRIPT_SUBMIT_DIR=$(scontrol show job "$SLURM_JOBID" | awk -F= '/Command=/{print $2}')
+  SCRIPT_SUBMIT_DIR="${SCRIPT_SUBMIT_DIR%% *}" # Removes anything starting at first space (so keep script path).
+else
+  SCRIPT_SUBMIT_DIR=$(realpath "$0")
```

Comment:
Could you motivate your change from `="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"` to `
=$(realpath "$0")`?

Reply:
To cite `info coreutils 'realpath invocation'`:
> 18.5 'realpath': Print the resolved file name.
> \==============================================
> 
> 'realpath' expands all symbolic links and resolves references to '/./',
'/../' and extra '/' characters.

So it's a cleaner solution code-wise (as it was created for these use-cases) that should also be able to resolve symbolic links (haven't tested this though). As it's part of [GNU coreutils](https://www.gnu.org/software/coreutils/), most modern Linux distro's should also support it by default (an otherwise installation should be rather easy).

There is a downside in that MacOS does not support it by default, though this is also the case for many other things such as `wget`. This can be fixed by using a package manager such as MacPorts or HomeBrew. For example, using MacPorts one can do `sudo port install coreutils` to install GNU coreutils and then add `export PATH=/opt/local/libexec/gnubin/:$PATH` to the `.bash_profile` to make GNU coreutils the default (instead of the BSD commands MacOS normally uses). Though again, this isn't a `realpath` specific issue but rather a MacOS issue in general.

Reply:
Sidenote: `realpath` is already used elsewhere in the script as well: https://github.com/molgenis/vip/blob/cf3954bc05ffff460eb7af7c64b724a03a54e252/pipeline.sh#L84